ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/AbstractBlockCreator.java

```diff
@@ -92,8 +96,10 @@ public AbstractBlockCreator(
     this.protocolSchedule = protocolSchedule;
     this.gasLimitCalculator = gasLimitCalculator;
     this.minTransactionGasPrice = minTransactionGasPrice;
+    this.minBlockOccupancyRatio = minBlockOccupancyRatio;
     this.miningBeneficiary = miningBeneficiary;
     this.parentHeader = parentHeader;
+    this.protocolSpec = protocolSchedule.getByBlockNumber(parentHeader.getNumber() + 1);
```

Comment:
How can we be sure that this block + 1 will always exist?

Reply:
Block + 1 does not have to exist. You can retrieve any procotol spec for any blocks whether or not they exist